jobs/kpi-forecasting/kpi_forecasting/models/prophet_forecast.py

```diff
@@ -20,15 +20,20 @@ class ProphetForecast(BaseForecast):
     def column_names_map(self) -> Dict[str, str]:
         return {"submission_date": "ds", "value": "y"}
 
-    def _fit(self, observed_df) -> None:
-        self.model = prophet.Prophet(
-            **self.parameters,
+    def _build_model(self, parameter_dict):
```

Comment:
What's the motivation for having `_build_model` in addition to `_fit`?

Reply:
The main reason is honestly just that it makes it easier to test, I can mock the function pretty easily (though I guess now that I think about it, it probably isn't that much harder to mock an attribute?)

I think it might also be nice in terms of thinking ahead to how we want to be able to drop in new types of models more easily, but the counterpoint to that is it's unlikely we can just modify `_build_model` without having to modify `_fit`